### ISSUE TITLE
Allow remove APIs set flags on items

### ISSUE THREAD

**Issue by Minoru** [newsboat/newsboat#1130]
Remove APIs like Miniflux etc. usually allow items to be "starred" or "favourited". In Newsboat, those marks turn into a specific flag, configured via e.g. `miniflux-flag-star`. But here's a problem (https://github.com/newsboat/newsboat/pull/1125#discussion_r466612243): when creating an `RssItem`, the remote API can't set a flag on it because:

1. `Cache::externalize_rssfeed` doesn't write `flags` into the database. This is an intentional and correct behaviour: the DB stores things that aren't part of RSS/Atom, like flags, read/unread status, deleted status etc. `Cache::externalize_rssfeed` stores only the fields that are present in RSS/Atom to avoid overwriting any of Newsboat's state.

  In short: calling `set_flags` on an item from a remote API does nothing, since those flags are not stored into the DB;

2. calling `RssItem::update_flags` after `RssItem::set_flags` *overwrites* existing flags instead of toggling the "starred" flag, since the `RssItem` on which the remote API operates doesn't come from a DB and doesn't "know" about the item's flags.

We need a way for a remote API to mark an item as "starred", and the rest of the system should *update* flags instead of overwriting them.

Note also that TT-RSS has *two* flags: "starred" and "publish". If at all possible, the new interface should be generic enough to allow multiple flags.
